source/agora/common/crypto/Schnorr.d

```diff
@@ -0,0 +1,413 @@
+/*******************************************************************************
+
+    Schnorr signature implementation using Ed25519
+
+    TODO:
+      - Add extended description
+      - Compress signature
+      - Check side channels attack on equality comparison (?)
+      - Move `Scalar` and `Point` to libsodiumd (?)
+      - Expose the keypair in a "nicer" format
+      - Add test for more than 2 party
+      - Provide nicer APIs
+      - Work on the `R` interaction
+
+    Super TODO:
+      - Audit for more attacks
+      - Audit GDC-generated code (/LDC?)
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.crypto.Schnorr;
+
+import agora.common.Data;
+import agora.common.Hash;
+
+import geod24.bitblob;
+import libsodium;
+
+import std.algorithm;
+import std.range;
+
+
+/// Single signature example
+nothrow @nogc unittest
+{
+    Pair kp = Pair.random();
+    auto signature = sign(kp, "Hello world");
```

Comment:
could also add:

```
auto sig2 = sign(kp, "Hello world");
assert(signature != sig2);  // prevents replay attacks
```

Reply:
I don't see where the replay attack is ?